File: Programming Basics/Programming Basics - C#/Exercises/05. Simple Loops/05. Simple Loops/ConsoleApp1/Odd - Even Position.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;


namespace ConsoleApp1
{
    class Program
    {
        static void Main(string[] args)
        {
            int n = int.Parse(Console.ReadLine());
            double oddSum = 0;
            double evenSUm = 0;
            double oddMin = double.MaxValue;
            double oddMax = double.MinValue;
            double evenMin = double.MaxValue;
            double evenMax = double.MinValue;
            double min = double.MaxValue;
            double max = double.MinValue;

           // int numOnoddposition = 0;
          //  int numOnevenPosition = 0;

            for (int raw = 1; raw <= n; raw++)
            {
                double num = double.Parse(Console.ReadLine());
                if (raw % 2 == 0)
                {
                    evenSUm += num;
                    if (num < evenMin)
                    {
                        evenMin = num;
                    }

                    if (num > evenMax)
                    {
                        evenMax = num;
                    }
                }
                else
                {
                    oddSum += num;
                    if (num < oddMin)
                    {
                        oddMin = num;
                    }

                    if (num > oddMax)
                    {
                        oddMax = num;
                    }
                }
                if (num < min)
                {
                    min = num;
                }

                if (num > max)
                {
                    max = num;
                }
            }
            Console.WriteLine("ODD sum = {0}", oddSum);
            Console.WriteLine("ODD Min = {0}", oddMin);
            Console.WriteLine("ODD Max = {0}", oddMax);
            Console.WriteLine("EVEN Sum = {0}", evenSUm);
            Console.WriteLine("EVEN Min = {0}", evenMin);
            Console.WriteLine("EVEN Max = {0}", evenMax);
            Console.WriteLine("Min = {0}", min);
            Console.WriteLine("<ax = {0}", max);
        }
    }
}

File: Programming Basics/Programming Basics - JavaScript/Exercises/05. Simple Loops/11. Odd Even Position/OddEvenPosition.js
function oddEvenPosition(args) {
    let n = Number(args[0]);
    let evenSum = 0;
    let oddSum = 0;
    let oddMin = Number.POSITIVE_INFINITY; //   1000000000.0
    let oddMax = Number.NEGATIVE_INFINITY; // - 1000000000.0
    let evenMin = Number.POSITIVE_INFINITY;//   1000000000.0
    let evenMax = Number.NEGATIVE_INFINITY;// - 1000000000.0

    for (let i = 1; i <= n; i++) {
        let num = Number(args[i]);

        if (i % 2 == 0) {
            evenSum += num;

            if (evenMax < num) {
                evenMax = num;
            }

            if (evenMin > num) {
                evenMin = num;
            }
        }
        else {
            oddSum += num;

            if (oddMax < num) {
                oddMax = num;
            }

            if (oddMin > num) {
                oddMin = num;
            }
        }
    }

    // for ODD numbers
    console.log(`Oddsum=${oddSum},`);
    //---------------------------------------------------------------------------------------------------------------------
    if (oddMin == Number.POSITIVE_INFINITY) {
        console.log("OddMin=No,");
    }
    else {
        console.log(`OddMin=${oddMin},`);
    }
    //---------------------------------------------------------------------------------------------------------------------
    if (oddMax == Number.NEGATIVE_INFINITY) {
        console.log("OddMax=No,");
    }
    else {
        console.log(`OddMax=${oddMax},`);
    }
    //---------------------------------------------------------------------------------------------------------------------
    // for EVEN numbers
    console.log(`EvenSum=${evenSum},`);
    //---------------------------------------------------------------------------------------------------------------------
    if (evenMin == Number.POSITIVE_INFINITY) {
        console.log("EvenMin=No,");
    }
    else {
        console.log(`EvenMin=${evenMin},`);
    }
    //---------------------------------------------------------------------------------------------------------------------
    if (evenMax == Number.NEGATIVE_INFINITY) {
        console.log("EvenMax=No,");
    }
    else {
        console.log(`EvenMax=${evenMax},`);
    }
}

/* Uncomment to test:
oddEvenPosition(["6", "2", "3", "5", "4", "2", "1"]);
oddEvenPosition(["2", "1.5", "2.5"]);
oddEvenPosition(["1", "1"]);
oddEvenPosition(["0"]);
    */